lib/hammer_cli_foreman_tasks/recurring_logic.rb

```diff
@@ -27,6 +27,30 @@ class CancelCommand < HammerCLIForeman::DeleteCommand
       build_options
     end
 
+    class ClearCancelledCommand < HammerCLIForeman::DeleteCommand
+      option '--cancelled', :flag, _("Only delete cancelled recurring logics")
+      option '--finished', :flag, _("Only delete finished recurring logics")
+      def request_params
+        params = super
+        raise ArgumentError, "Please specify if you want to remove cancelled or finished recurring logics using --cancelled or --finished " unless (options['option_cancelled'] || options["option_finished"])
+        cancelled = "state=cancelled" if options["option_cancelled"]
+        finished = "state=finished" if options["option_finished"]
+        params["search"] = cancelled || finished
+        params
+      end
+
+      action :bulk_destroy
+      command_name 'clear-cancelled'
+      desc _("Delete cancelled reccuring logics")
+      success_message _('Recurring logics deleted.')
+      failure_message _('Could not delete  recurring logics')
+
+      def execute
+        puts send_request.to_s
+        HammerCLI::EX_OK
+      end
```

Comment:
Added this so the user can see what was deleted but I'm not sure if its needed

Reply:
@xprazak2 @ofedoren opinions on this?

Reply:
I think it might be a good idea to provide some information, but currently I'm not able to provide the best way to show the info, because don't have any..

BTW, if nothing was deleted/cleared the empty array is returned and thus is being printed, which I'm finding not the best UX.

Reply:
Added a message if array is empty